src/FSharpLint.Core/Framework/Ast.fs

```diff
@@ -386,7 +386,15 @@ module Ast =
             add <| expressionNode expression2
             add <| expressionNode expression1
             add <| expressionNode expression
-        | SynExpr.LetOrUseBang(_, _, _, pattern, expression, expression1, _)
+        | SynExpr.LetOrUseBang(_, _, _, pattern, rightHandSide, andBangs, leftHandSide, _) ->
+            add <| expressionNode rightHandSide
+            add <| expressionNode leftHandSide
+            // TODO: is the the correct way to handle the new `and!` syntax?
+            andBangs |> List.iter (fun (_, _, _, pattern, body, _) ->
```

Comment:
the and! syntaxes are of type `list<SequencePointInfoForBinding * bool * bool * SynPat * SynExpr * range>` here, so the interesting things are the pattern and the body, which I've extracted here.

I'm not sure this is correct, though.

Reply:
Sorry for the slow response on this, I've had a quick look this morning before work and it looks good, but I do notice all the others are added in reverse order. Have to admit I can't actually recall why, I'll take a look after work and find out how important it is for them to be in order, and if so get a test to cover it